### PR TITLE
[DS-4167] Migrate update-sequences.sql to `database` command

### PR DESCRIPTION
**Pull request by terrywbrady**
Fixes #7509

Migrate update-sequences.sql to the _dspace database_ command.

In addition to being convenient, this will enable dspace docker instances populated from AIP content to have their sequences automatically updated.

```
$ docker exec dspace //dspace/bin/dspace database update-sequences
Running org/dspace/storage/rdbms/sqlmigration/postgres/update-sequences.sql
update-sequences complete
```

Since the dspace/etc/ directory will be going away, move the SQL to dspace-api/src/main/resources to reside with the Flyway SQL.

### Migration needed
- 6x
- 5x
- 4x (maybe)

### Input needed
- Should this PR remove dspace/etc directory?
- How can the existing Oracle update-sequences.sql be converted to a single SQL file?  The current solution requires the SQL to reside on the file system.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java [hunk 1]
```diff
@@ -328,24 +330,54 @@ public static void main(String[] argv) {
...
+                    String sqlfile = "org/dspace/storage/rdbms/sqlmigration/" + dbType +
+                            "/update-sequences.sql";
+                    InputStream sqlstream = DatabaseUtils.class.getClassLoader().getResourceAsStream(sqlfile);
+                    if (sqlstream != null) {
+                        StringBuilder sb = new StringBuilder();
+                        try (BufferedReader br = new BufferedReader(new InputStreamReader(sqlstream))) {
+                            for (String line = br.readLine(); line != null; line = br.readLine()) {
+                                sb.append(line);
+                            }
+                        }
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java]: You should be able to replace all this fancy string parsing with just one line:
`String sqlString = IOUtils.toString(sqlstream, "UTF-8");`

That line will convert the UTF-8 InputStream into a String.

IOUtils requires an `import org.apache.commons.io.IOUtils`.

**Comment by tdonohue** [comment]
@terrywbrady : I gave this a quick look today, and it looks like a good start. I added one inline comment/suggestion.

Regarding your questions in the description:
* I think we should remove the `dspace/etc` directory. After this change it only contains a README.  That README was only there to note that the SQL scripts moved between 4.x and 5.x, and it's already duplicated by documentation (on the wiki).
* You likely won't be able to port this to 4.x, as the `dspace database` commandline tool didn't exist until 5.x. Honestly, I'm a little hesitant to port this back to prior releases, as we don't tend to move around files like this during minor upgrades.  If you need it ported to 5.x/6.x for Docker, maybe we consider leaving the `dspace/etc` folder in 5.x/6.x (and leaving the update-sequence.sql file in that location), so that we don't confuse 5.x/6.x users after their next minor upgrade.  I like this PR though as-is for the 7.x version.
* I don't have an answer on the Oracle question -- hopefully someone with more Oracle experience could give some tips.

**Comment by mwoodiupui** [comment]
You may want to look at making `updateseq.sql` into a PL/SQL subprogram, tucking its declaration into `update-sequences.sql`.
BTW the internal documentation for updateseq seems to be incomplete, as it doesn't describe the fourth parameter used with the `handle_seq` table.

File: dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java [hunk 2]
```diff
@@ -328,24 +330,54 @@ public static void main(String[] argv) {
                     e.printStackTrace();
                     System.exit(1);
                 }
+            } else if (argv[0].equalsIgnoreCase("update-sequences")) {
+                try (Connection connection = dataSource.getConnection()) {
+                    String dbType = getDbType(connection);
+                    String sqlfile = "org/dspace/storage/rdbms/sqlmigration/" + dbType +
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java]: Should this not be either absolute "/org/dspace/storage/rdbms/sqlmigration/" or relative "sqlmigration/"?

File: dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java (hunk 1 above)
Comment (terrywbrady) [reply dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java]: Added commit 22cc5469a23

File: dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java (hunk 2 above)
Comment (terrywbrady) [reply dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java]: The other worked, but it seems more clear to start with /.  Added commit e421a7c7c75

**Comment by terrywbrady** [comment]
@mwoodiupui, I added your recommended Oracle stored procedure.  I haven't yet figured out if I will be able to test this in Docker.  Hopefully we can find someone who can verify the change in a DSpace Oracle DB.

@tdonohue , I have removed the etc directory.  Your recommendations on the port to 5x and 6x make sense.  The process will need to read from the filesystem rather than the classpath in order to execute then change.  I will likely leave the Oracle SQL alone for 5x/6x.  I will tackle that after this code is merged.

File: dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java (hunk 2 above)
Comment (terrywbrady) [reply dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java]: The resource is not found if it is prefixed with /.  Perhaps that is treated as a filesystem path.  Reverting to "org/..."

**Comment by terrywbrady** [comment]
@mwoodiupui , your SQL is working as a part of the script.
```
 MINGW64 ~/DSpace-Docker-Images/docker-compose-files/dspace-oracle-compose (master)
$ docker ps
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS              PORTS                              NAMES
c80cee226a93        dspace/dspace:local-build   "catalina.sh run"        25 seconds ago      Up 7 seconds        8009/tcp, 0.0.0.0:8080->8080/tcp   dspace
8aff80560ae9        atmire/dspace-oracle:11g2   "/entrypoint.sh /bin…"   26 seconds ago      Up 25 seconds       1521/tcp, 8080/tcp                 dspacedb

~/DSpace-Docker-Images/docker-compose-files/dspace-oracle-compose (master)
$ docker exec dspace //dspace/bin/dspace database update-sequences
Running org/dspace/storage/rdbms/sqlmigration/oracle/update-sequences.sql
update-sequences complete
```

**Comment by terrywbrady** [comment]
@mwoodiupui , @tdonohue , could you give this a quick re-review?  I think I addressed your comments.

**Review by tdonohue — APPROVED** [review]
👍 The code looks fine to me. However, I haven't had a chance to test this on Postgres or Oracle.

**Review by mwoodiupui — APPROVED** [review]
Looks right, works on Oracle, works on PostgreSQL.

**Pull request merged**